.github/workflows/reusable-build-docsite.yaml

```
@@ -0,0 +1,28 @@
+name: Build the Ansible docsite
+
+on:
+  workflow_call:
+
+env:
+  PIP_DISABLE_PIP_VERSION_CHECK: 1
+
+jobs:
+  build-docsite:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install requirements
+        run: python -m pip install -r requirements.in -c requirements.txt
+      - name: Generate assets for the docs landing page
+        run: python -I -Werror build.py
```

To emulate strict warnings mode, try this:
```suggestion
      - name: Install requirements
        run: python -Im pip install nox
      - name: Generate assets for the docs landing page
        env:
          PYTHONWARNINGS: error
        run: python -Im nox -s build
```
(although, I'm not sure if `nox` bypasses the env vars into the internal invocation)

---

That should work:
> By default, all environment variables are passed.

_(https://nox.thea.codes/en/stable/config.html#nox.sessions.Session.run)_

---

Hmm. So "PYTHONWARNINGS: error" does what exactly here? As you said there is the nox internal invocation so this seems like it would just treat errors with nox itself as exceptions right?

Why not just `nox -s build` here?

---

Ah, wait. I see it now. Nevermind my previous comment, thanks again.